packages/jellyfish-api-core/__tests__/category/icxorderbook/createOrder.test.ts

```diff
@@ -0,0 +1,225 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import {
+  ICXGenericResult, ICXOfferInfo, ICXOrderInfo, ICXOrder, UTXO, ICXOrderStatus, ICXOrderType
+} from '../../../src/category/icxorderbook'
+import BigNumber from 'bignumber.js'
+import { accountBTC, accountDFI, ICXSetup, idDFI, symbolBTC, symbolDFI } from './icx_setup'
+import { RpcApiError } from '../../../src'
+
+describe('ICXOrderBook.createOrder', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+  const icxSetup = new ICXSetup(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await icxSetup.createAccounts()
+    await icxSetup.createBTCToken()
+    await icxSetup.initializeTokensIds()
+    await icxSetup.mintBTCtoken(100)
+    await icxSetup.fundAccount(accountDFI, symbolDFI, 500)
+    await icxSetup.fundAccount(accountBTC, symbolDFI, 10) // for fee
+    await icxSetup.createBTCDFIPool()
+    await icxSetup.addLiquidityToBTCDFIPool(1, 100)
+    await icxSetup.setTakerFee(0.001)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  afterEach(async () => {
+    // cleanup code here
+  })
+
+  it('should createOrder to sell 15 dfi', async () => {
```

Comment:
```suggestion
  it('should createOrder to sell 15 DFI from chain:DFI to chain:BTC', async () => {
```

Reply:
at the moment we only have DFI and BTC. So it's indicative.
But forward-thinking, better to add `from` and `to` chains.
Btw I got the idea of no capitals other than code components from this comment -> https://github.com/DeFiCh/jellyfish/pull/398#discussion_r655920165
I searched online for JS BDD style naming conventions and got mixed results.
Is there a reference to the naming style that we follow here? would be easy. @fuxingloh

Reply:
For specifying the chain, the main reason is BTC can exist in the defichain and bitcoin chain. Easier to understand which chain you are swapping from and to.

For BDD, my previous comment was more towards; whether you borrow a testing style from another framework or language because you are consistent within your PR. But this repo follows another consistency structure.

Frankly speaking, the style that I am following is more towards a linguistic conversational writing perspective. Why I don't use capitalization is because we aren't actually talking/testing in complete sentences. More towards a loose exchange of ideas in a listicle sense.

1. describe('a scenario')
    1. it('should do something')
    1. it('should do another thing')
1. describe('another scenario')
    1. it('should do something')

> I will put this into `CONTRIBUTING.md` when I have the time https://github.com/DeFiCh/wallet/blob/main/CONTRIBUTING.md

For reference for you to see what the pattern is like; other tests that the team wrote in other context/concerns:

1. https://github.com/DeFiCh/whale/blob/main/src/module.database/provider.level/level.database.spec.ts
2. https://github.com/DeFiCh/whale/blob/main/src/module.api/cache/defid.cache.spec.ts
3. https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_poolswap.test.ts
4. https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_utxo.test.ts
5. https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-transaction-signature/__tests__/tx_signature/sign_transaction.test.ts
6. https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-transaction-signature/__tests__/tx_signature/sign_input.test.ts
7. https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-crypto/__tests__/elliptic.test.ts

Reply:
ok. had some doubts. now clear. thanks!